android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1003,4 +1861,21 @@ public void onTileCountChanged() {
     public void onCryptoWidgetBottomSheetDialogDismiss() {
         startTimer();
     }
+
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        mBraveNewsController = null;
+        InitBraveNewsController();
+    }
+
+    private void InitBraveNewsController() {
+        if (mBraveNewsController != null) {
+            return;
+        }
+
+        mBraveNewsController =
```

Comment:
`mBraveNewsController.close()` in onDestroy

Reply:
would it be good enough to put it in `protected void onDetachedFromWindow()`.  this class is extended from `NewTabPageLayout` and that has` onDestroy` as private

Reply:
yeah I think that will work